### DISCUSSION TITLE
BPF Syntex Error

### DISCUSSION THREAD

**Discussion by baileyphil** [Security-Onion-Solutions/securityonion#10646] [2.4] [answered]
Afternoon, 

Was wondering if somebody could help me out please, relatively new to Security Onion, and im just trying to clean up some of the alerts, i have added a BPF filter but when i reload Suricata im getting a syntax error

----------
          ID: suribpfcompilationfailure
    Function: test.configurable_test_state
      Result: False
     Comment: BPF Syntax Error - Discarding Specified BPF
     Started: 10:39:48.476296
    Duration: 11.437 ms
     Changes:

This is the entries I have in my global.sls file

    164 nids:
    165   bpf:
    166   - "not host 192.168.10.16"
    167   - "not (src net 192.168.0.0/24 && dst host 192.168.13.26 && dst port 1433)"

When i initially did it i just put the 166 line in saved it and it seemed to work, then i did a return double space and added the next line, thats when i got the syntax error.

Im sure im doing something very basic wrong, but i have had a look though the discussion boards and cant seem to find a solution.

Any help for a noob is very much appreciated

Thanks 

Phil :)

**Comment by dougburks** [comment]
Have you tried adding `and` to the end of the first BPF entry?
https://docs.securityonion.net/en/2.4/bpf.html#multiple-conditions

**Comment by baileyphil** [comment]
Morning Doug,

Thanks for the response much appreciated, i did as you recommended pull all the text onto a single line and put an "and" in-between.

164 nids:
165   bpf:
166   - "not host 192.168.10.16" and "not (src net 192.168.0.0/24 && dst host 192.168.13.26 && dst port 1433)"

However when restarting the server all the services went into a "Waiting to Start" state.

I removed the and and everything after it, restarted and everything is now fine, i assume im missing something in the formatting?

Sorry i have tried a few different formatting and wordings but it doesn't seem to make any difference.

Any help is appreciated.

Thanks 

Phil

**Reply by dougburks** [reply, marked as answer]
You posted this in the 2.4 category. Are you using version 2.4 (currently in Beta) or are you still using 2.3?

Have you tried it like this?
```
not host 192.168.10.16 and not (src net 192.168.0.0/24 and dst host 192.168.13.26 and dst port 1433)
```

**Comment by baileyphil** [comment]
I think this has worked, Suricata certainly seems to have restarted and accepted the changes, thank you Doug for your help much appreciated.

So if i say wanted to add another is it just a case of doing a "and not (dst host 192.168.16.11 and dst port 22" to the end of that line?

**Reply by dougburks** [reply]
> So if i say wanted to add another is it just a case of doing a "and not (dst host 192.168.16.11 and dst port 22" to the end of that line?

Yes, that should be correct.
